README.md

```diff
@@ -1,2 +1,156 @@
+    # ${{ secrets.GITHUB_TOKEN }}.
+    # Used to fetch your repository's pull requests via Octokit.
+    # 
+    # Default: ${{ github.token }}
+    github-token: ''
+
+    # The perennial branches for your repository.
+    #
+    # Only necessary when not using a shared Git Town config file, as this
+    # extension will read the repository's perennial branches from config
+    # when possible. 
+    perennial-branches: |
+
+    # The regex used by the action to find and replace the visualization
+    # within a PR's body.
+    # 
+    # Default: '[ \t]*?## Stack[\s\S]*?(?=##|$)'
+    body-regex: ''
+
+    # The template used to generate a visualization of stacked changes for
+    # the current pull request. Template MUST contain `{{ stack }}` inside.
+    # 
+    # Default: |
+    #   ## Stack
+    #   {{ stack }}
+    body-template: |
+
+```
+<!-- end usage -->
+
+# Guides
+
+## Getting Started
+
+To get started, create a workflow file in `.github/workflows` with the following
+configuration:
+
+```yaml
+name: Git Town
+
+on:
+  pull_request:
+    - **
+
+jobs:
+  git-town:
+    name: Sync Visualization
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: git-town/action@v1
+        with:
+          github-token: ${{ github.token }}
+```
+
+Anytime a pull request is created or updated, the action will visualize
+your stacked changes for all PRs in the stack, like so:
+
+- `main`
+  - #1 :point_left:
+    - #2
+
+## Customizing the Visualization
+
+The `body-regex` input is used by this extension to find and replace
+content inside of the PR body for each of your stacked changes.
+The content that it replaces the match with is determined by the
+`body-template` input. For example:
+
+```md
+## Stack   ┐
+- `main`   ├ This is what is replaced by the default body regex.
+  - #1     │
+     - #2  │
+           ┘
+## Description
+
+This is a stacked change.
+```
+
+To customize the 
+
+```diff
+- ## Stack
+- - `main`
+-   - #1
+-      - #2
++ ## Stack
++ - `main`
++   - #1
++      - #2
++        - #3
++
+## Description
+
+...
+```
+
+Since the action replaces the _entire_ section (including the header), it
+needs to know the structure of the section so that it can rebuild it. This
+is where the body template comes in.
+
+The body template is simply a string with `{{ stack }}` somewhere inside it.
+The action will replace `{{ stack }}` with the visualization for each PR in
+a stack. Based on the example above, the correct body template would be:
+
+```md
+## Stack
+{{ stack }}
+```
+
+With all of this in mind, this is an example of using `body-regex` and `body-template`
+to change the heading name and level that is used by this action:
+
+```yaml
+- uses: git-town/action@v1
+  with:
+    # Change the header level & name
+    body-template: |
+      # Stacked Changes
+      {{ stack }}
+
+    # Update the regex to look for this header
+    body-regex: '[ \t]*?# Stacked Changes[\s\S]*?(?=#|$)'
```

Comment:
Question around this regex: AFAIK `[\s\S]` includes both whitespace and non-whitespace characters, i.e. matches all characters. Could it be replaced with `.`?

Reply:
Good question! Unfortunately not as `.` does not match newline characters, meaning the match would end after the first line.

Reply:
Putting the regex in [dotall mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/dotAll) might help makes the `.` match newlines.